### ISSUE TITLE
How I generate an array of single values dynamically?

### ISSUE THREAD

**Issue by avaz** [BrynCooke/fluent-json#1]
Hi,

I can't figure out how to generate an array of single values dynamically with this library?

What I want is generate a json like:

`{"name": "Joe", "tastes": ["chicken", "pasta"]}`

My Entity is like:

```
public class User {
  private long id;
  private String name;
  private List<Taste> tastes;
  // getters/setters
}
public class Taste {
  private long id;
  private String description;
}
```

I'm trying to build the json in the following way:

```
JsonObject json = buildObject()
      .add( "name", user.getName() )
      .add( "tastes", buildArray( (t) -> new JsonBuilder() {
        @Override
        public void write( Writer out ) throws IOException {
          out.write( t.getDescription() );
        }

        @Override
        public void write( JsonWriter out ) throws IOException {
          out.value( t.getDescription() );
        }
      }, user.getTastes() ) )
      .end()
      .getJson();
```

But it generates the a ClassCastException:

`java.lang.ClassCastException: test.GenJson$1 cannot be cast to org.jglue.fluentjson.JsonBuilderFactory$Impl`

Seems to be a bug in the library as the builders can only be of type `Impl` that is a private class of `JsonBuildFactory`. But I can say it for sure.

Can you help?

Regards!

**Comment by BrynCooke** [comment]
```
List<String> likes = Arrays.asList("chicken", "pasta");
        JsonObject json = buildObject()
                  .add( "name", "Joe" )
                  .add( "tastes", buildArray().forEach((a,v)->a.add(v), likes))
                  .getJson();
```

It's been a while since I looked at this library.
Dynamic building support could definitely be better. Perhaps I'll get round to it some day.

**Comment by avaz** [comment]
Hi Bryn,

The code that you suggest is not working. I'm using the latest version of `fluent-json`, `2.0.1` and there is no `forEach` method in `buildArray()` (`JsonArrayBuilder`). 
Do you have any idea?

Thanks!

**Comment by BrynCooke** [comment]
Hi
Apologies. I had been using code from my local branch.
I've had a think and released 2.0.3 where the following code works.

```
        List<String> likes = Arrays.asList("chicken", "pasta");
        JsonObject json = buildObject()
                  .add( "name", "Joe" )
                  .add( "tastes", buildArray().addAll(Mapper.STRING, likes))
                  .getJson();
```

**Comment by avaz** [comment]
Hey Bryn, now its working! 
Thanks!
